server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -107,8 +108,58 @@
 import com.beust.jcommander.Parameter;
 import com.google.common.base.Preconditions;
 
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+
 public class Compactor extends AbstractServer implements CompactorService.Iface {
 
+  private static class CompactionProgress implements MetricsProducer {
+    private final long warnTime;
+    private long time;
+    private long read;
+    private long written;
+    private boolean stuck = false;
+
+    public CompactionProgress(AccumuloConfiguration config) {
+      warnTime = config.getTimeInMillis(Property.TSERV_COMPACTION_WARN_TIME);
+    }
+
+    @Override
+    public void registerMetrics(MeterRegistry registry) {
+      Gauge.builder(METRICS_TSERVER_MAJC_STUCK, this, CompactionProgress::getNumStuck)
```

Comment:
With modification - this could show the "stuck" time rather than a boolean.  This would allow users to set external alarms based on time rather than on a boolean condition. It could also provide insight if things can somehow become "unstuck" without intervention, you could plot the stuck time over time to see how things are progressing.

Reply:
So, each Compactor runs one majc at a time, but the TabletServer can run many of them at once. I was approaching this from the perspective that a user could just plot stuck majc's over time and be able to drill down to one of the processes. For your suggestion to work, I think I would need to emit multiple Timers , one per stuck extent?

Reply:
From looking at line 134 - it seemed that the value would only be 0 or 1 - and that there is a summation occurring later?

It seemed that more info could be conveyed other than a boolean. If it is being summed elsewhere, does it need to be a metric here?

I could just be reading this wrong...

Reply:
Right, the Compactor only runs one majc at a time. So, the current one is either stuck, or not stuck. The summation across all processes (or by some other aggregation) would be done in a downstream time series database.

Reply:
+1 to the summation across all.

I think I am just triggered by the compactor count of 0 or 1 being a metric.  I guess I'm asking: Can it be kept as a count - but not as a metric? Or, if kept as a metric provide additional information?

Reply:
I'm confused by your use of `metric` in the latest comment. I have it set as a `gauge` because the number of stuck compactions is going to go up and down, in both the TabletServer and Compactor, over time.

Reply:
I'm using metric as - a value that is reported by the metrics system

Reply:
> Can it be kept as a count - but not as a metric? Or, if kept as a metric provide additional information?

So, if it's not a value reported by the metrics system, then what do you mean by keeping it as a `count` ? What additional information do you think need needs to be conveyed?

Reply:
Check out the changes in bec830ff3d0dc606864e5172fc463f80224f5ca7, I modified the code to use a LongTaskTimer.